### PR TITLE
Update dev docs after setuptools update

### LINKED ISSUE #37934: Update dev debugging docs
**Describe the outcome that is desired.**
Since #37775 was merged, the instructions for debugging Workbench need to be updated. VS was fixed in #37925 so probably no documentation will need updating for that, just PyCharm, macOS, and Linux.

### PR DESCRIPTION
**Pull request by thomashampson**
### Description of work
Update dev docs after the setuptools deprecation warnings were fixed [here](https://github.com/mantidproject/mantid/pull/37775).

Fixes #37934 

### To test:
- Make sure the new instructions work on all OS.
- Make sure the docs are correctly formatted.

*This does not require release notes* because **it's a dev-docs update**

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

File: dev-docs/source/CLion.rst [hunk 1]
```diff
@@ -119,23 +119,35 @@ Debugging with CLion
...
 
-#. Set the program arguments
+#. Set the program arguments:
 
    .. hlist::
       :columns: 1
 
-      - On Linux: ``workbench --single-process``
+      - On Linux & macOS: ``workbench --single-process``
       - On Windows: ``workbench-script.pyw --single-process``
```
Comment (thomashampson) [inline dev-docs/source/CLion.rst:136]: I think this will have to change

**Comment by jmborr** [comment]
@thomashampson I included some additional documentation on building with CLion for the optional scenario of starting CLion from a terminal without having to first activate the `mantid-developer` conda environment

**Review by jhaigh0 — DISMISSED** [review]

File: dev-docs/source/CLion.rst [hunk 2]
```diff
@@ -119,23 +130,35 @@ Debugging with CLion
...
 
-#. Set the program arguments
+#. Set the program arguments:
 
    .. hlist::
       :columns: 1
 
-      - On Linux: ``workbench --single-process``
-      - On Windows: ``workbench-script.pyw --single-process``
+      - On Linux & macOS: ``workbench --single-process``
```
Comment (SilkeSchomann) [inline dev-docs/source/CLion.rst:146]:
```suggestion
      - On Linux & macOS & Windows: ``workbench --single-process``
```
Then the second line ("On Windows") could just get removed.

Comment (adriazalvarez) [reply dev-docs/source/CLion.rst:146]: Sorry for this last minute comment, on my CLion configuration on mac, mantid only launches  from the cmake configuration I set program arguments to : ``-m workbench --single-process`` . 
To make it work as in pycharm, is necessary to create a python run configuration.

Comment (thomashampson) [reply dev-docs/source/CLion.rst:146]: Thanks for that @adriazalvarez . I don't use CLion so am not best placed to make these changes. Would you be able to add a commit to implement that?

Comment (adriazalvarez) [reply dev-docs/source/CLion.rst:146]: @thomashampson I only use it on MacOS, I don't know if they same applies to other systems.

File: dev-docs/source/CLion.rst [hunk 3]
```diff
@@ -119,23 +130,35 @@ Debugging with CLion
...
+      - On Windows: ``workbench --single-process``
```
Comment (adriazalvarez) [inline dev-docs/source/CLion.rst:147]:
```suggestion
      - On Linux, macOS and Windows: ``-m workbench --single-process``
```

**Review by jhaigh0 — APPROVED** [review]

**Pull request merged**